pkg/hintrunner/zero/zerohint_math_test.go

```diff
@@ -193,5 +193,154 @@ func TestZeroHintMath(t *testing.T) {
 				check: apValueEquals(feltUint64(1)),
 			},
 		},
+
+		"IsPositive": {
+			{
+				operanders: []*hintOperander{
+					{Name: "value", Kind: apRelative, Value: feltInt64(10)},
+					{Name: "is_positive", Kind: unitialized},
+				},
+				makeHinter: func(ctx *hintTestContext) hinter.Hinter {
+					return newIsPositiveHint(ctx.operanders["value"], ctx.operanders["is_positive"])
+				},
+				check: varValueEquals("is_positive", feltInt64(1)),
+			},
+
+			{
+				operanders: []*hintOperander{
+					{Name: "value", Kind: apRelative, Value: feltInt64(0)},
+					{Name: "is_positive", Kind: unitialized},
+				},
+				makeHinter: func(ctx *hintTestContext) hinter.Hinter {
+					return newIsPositiveHint(ctx.operanders["value"], ctx.operanders["is_positive"])
+				},
+				check: varValueEquals("is_positive", feltInt64(1)),
+			},
+
+			{
+				operanders: []*hintOperander{
+					{Name: "value", Kind: apRelative, Value: feltInt64(-1)},
+					{Name: "is_positive", Kind: unitialized},
+				},
+				makeHinter: func(ctx *hintTestContext) hinter.Hinter {
+					return newIsPositiveHint(ctx.operanders["value"], ctx.operanders["is_positive"])
+				},
+				check: varValueEquals("is_positive", feltInt64(0)),
+			},
+
+			{
+				operanders: []*hintOperander{
+					{Name: "value", Kind: apRelative, Value: feltAdd(&utils.FeltMax128, feltInt64(-1))},
+					{Name: "is_positive", Kind: unitialized},
+				},
+				makeHinter: func(ctx *hintTestContext) hinter.Hinter {
+					return newIsPositiveHint(ctx.operanders["value"], ctx.operanders["is_positive"])
+				},
+				check: varValueEquals("is_positive", feltInt64(1)),
+			},
+
+			{
+				operanders: []*hintOperander{
+					{Name: "value", Kind: apRelative, Value: &utils.FeltMax128},
+					{Name: "is_positive", Kind: unitialized},
+				},
+				makeHinter: func(ctx *hintTestContext) hinter.Hinter {
+					return newIsPositiveHint(ctx.operanders["value"], ctx.operanders["is_positive"])
+				},
+				check: varValueEquals("is_positive", feltInt64(0)),
+			},
+
+			{
+				operanders: []*hintOperander{
+					{Name: "value", Kind: apRelative, Value: feltAdd(&utils.FeltMax128, feltInt64(1))},
+					{Name: "is_positive", Kind: unitialized},
+				},
+				makeHinter: func(ctx *hintTestContext) hinter.Hinter {
+					return newIsPositiveHint(ctx.operanders["value"], ctx.operanders["is_positive"])
+				},
+				check: varValueEquals("is_positive", feltInt64(0)),
+			},
+		},
+
+		"SplitIntAssertRange": {
+			// Zero value - assertion failed, any other - nothing.
```

Comment:
If the comment I made above about the implementation of this hint is correct, then these tests should be fixed as well

Reply:
Done